astropy/io/fits/tests/test_header.py

```diff
@@ -2388,6 +2388,39 @@ def test_set_keyword_with_space(self):
         assert hdr['KEY2'] == 4
         assert hdr['KEY2 '] == 4
 
+    def test_update_invalid_card(self):
+        """
+        Regression test for https://github.com/astropy/astropy/issues/5408
+
+        Tests updating the value of a card that is malformatted (with an
+        invalid value literal).
+
+        This tests two ways of reproducing the problem, one working with a
+        Card object directly, and one when reading/writing a header containing
+        such an invalid card.
+        """
+
+        card = fits.Card.fromstring('KW      = INF  / Comment')
+        card.value = 'FIXED'
+        assert tuple(card) == ('KW', 'FIXED', 'Comment')
+        card.verify('fix')
+        assert tuple(card) == ('KW', 'FIXED', 'Comment')
+
+
```

Comment:
To fix the code style check:
```suggestion

```

Reply:
Oops.  I should really set up my editor to fix that.